### DISCUSSION TITLE
Should we have a `putOrCreateValue` method?

### DISCUSSION THREAD

**Discussion by ghost** [kessler/node-regedit#89] [Features and Ideas]
`Error: registry path does not exist`

Got this error when I tried to put key value on .`.\\Policies\\System`, because Policies folder is empty. Is there a way to check if the folder/path existed and if no, create it?

**Comment by kessler** [comment]
for now you'll have to `list` it before to check. Might implement this in the future

**Comment by kessler** [comment]
@M1cr0M1nd following our discussion the other day I checked and we are *not* lazily creating new paths in the registry if they are missing. Also, the underlying vbs registry api does not provide a flag or some other way to do this lazily and I don't think it make sense to implement this in our VBS layer of JS layer as default behavior of `putValue`.

We might consider have a separate function for this, eg: `putOrCreateValue` or something along these lines, but we should really see if there's a demand for this at all...
